lib/rubocop/cop/mixin/hash_slice_except.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Common functionality for Style/HashExcept and Style/HashSlice
+    module HashSliceExcept
```

Comment:
Maybe something like `HashSubset` would be a better name? Also could you elaborate on the top comment to explain what this mixin does?